cpp/src/arrow/ipc/json_simple_test.cc

```diff
@@ -185,6 +189,21 @@ class TestIntegers : public ::testing::Test {
 
 TYPED_TEST_SUITE_P(TestIntegers);
 
+template <typename DataType>
+std::vector<typename DataType::c_type> TestIntegersMutateIfNeeded(
+    std::vector<typename DataType::c_type> data) {
+  return data;
+}
+
+// TODO: This works, but is it the right way to do this?
```

Comment:
I'm curious, what is this function meant for?

Reply:
I should remove this since we probably don't need to run this test on `HalfFloatType`.